### PR TITLE
Standardizing and Enhancing S3 connector Documentation

### LINKED ISSUE #484: Enhance S3 Data Connector Documentation
- [x] Connector description
- [x] Configuration section
- [x] Secrets section / Authentication section
- [x] Examples section
- [x] Parameters table
- [x] File format support (if applicable)

### PR DESCRIPTION
**Pull request by eadgbear**
## 🗣 Description

Standardizing and enhancing documentation as outlined in #580 

- **Standardize and enhance the S3 connector documentation**
- **Removing 'your'**
- **Typo fix, removing excess detail**
- **Conciseness and clarity pass**

## 🔨 Related Issues

Closes #484

### CODE REVIEW

File: spiceaidocs/docs/components/data-connectors/s3.md [hunk 1]
```diff
@@ -212,7 +174,7 @@ Create a dataset named `cool_dataset` from a Parquet file stored in MinIO.
     s3_region: 'us-east-1' # Best practice for Minio
 ```
 
-### S3 Public Example
+### Public bucketExample
```
Comment (Sevenannn) [inline spiceaidocs/docs/components/data-connectors/s3.md]: `Public Bucket Example`

File: spiceaidocs/docs/components/data-connectors/s3.md [hunk 2]
```diff
@@ -81,125 +79,89 @@ For non-public buckets, `s3_auth: key` or `s3_auth: iam_role` is required. `s3_a
...
+datasets:
+  - from: s3://s3-bucket-name/path/to/parquet/cool_dataset.parquet
+    name: cool_dataset
+    params:
+      s3_region: us-east-1
+      s3_key: ${s3:key}
+      s3_secret: ${s3:secret}
+```
+
+Learn more about [Kubernetes Secret Store](/components/secret-stores/kubernetes).
```
Comment (Sevenannn) [inline spiceaidocs/docs/components/data-connectors/s3.md]: Should we keep the keyring example as well?

File: spiceaidocs/docs/components/data-connectors/s3.md [hunk 3]
```diff
@@ -4,41 +4,22 @@ sidebar_label: 'S3 Data Connector'
 description: 'S3 Data Connector Documentation'
 ---
 
-import Tabs from '@theme/Tabs';
-import TabItem from '@theme/TabItem';
-
+Amazon S3 (Simple Storage Service) is a scalable object storage service that allows users to store, manage, and retrieve any amount of data from anywhere on the web.
```
Comment (lukekim) [inline spiceaidocs/docs/components/data-connectors/s3.md]: I don't think we need an intro for S3, given if you are looking at this page, and you have an S3 data source, you would likely know what it is.

**Review by Sevenannn — APPROVED** [review]

File: spiceaidocs/docs/components/data-connectors/s3.md (hunk 1 above)
Comment (eadgbear) [reply spiceaidocs/docs/components/data-connectors/s3.md]: Fixed

**Pull request merged**